BlazorApp1/wwwroot/blazor-fusionCharts.js

```diff
@@ -1,33 +1,72 @@
+
+const FUNCTION_KEYWORD = 'function()';
+
+// reviver function, key is passed by the JSON.parse
 function parseFunction(key, value) {
-        if (typeof (value) === 'string' && value.includes('function()')) {
-            let regEx = new RegExp("{([^}]*)}");
-            let func = new Function(value.split("function()")[1]);
-            return func;
-        }
-        return value
-    }
-    // Responsible for rendering any chart//
-    window.FusionCharts.renderChart = (chartConfiguration) => {
-        options = JSON.parse(chartConfiguration, this.parseFunction);
-        const chart = new FusionCharts(options);
-        chart.render();
-    };
-    //Generic Method to call any fusion chart method exluding except above methods//
-    window.FusionCharts.invokeChartFunction = (functionName, chartID, ...args) => {
-        var currentChart = FusionCharts(chartID);
-        var result =   currentChart[functionName].apply(currentChart, ...args);
-        var typeofresult = typeof result;
-        if(typeofresult === "number" || typeofresult === "boolean"){
-            result = result.toString();
-        }
-        else if(typeofresult === "object"){
-            result = JSON.stringify(result);
-        }
-        else if(typeofresult === "XML"){
-            result = XML.stringify(result);
-        }
-        if(typeofresult === "string"){
-            return result;
-        }
-        return result.toString() || "";
-    };
\ No newline at end of file
+  if (typeof value === "string" && value.includes(FUNCTION_KEYWORD)) {
+    let functionParts = value.split("function()")
+    let functionBody = functionParts[1]
+    let parsedFn = new Function(functionBody);
+    return parsedFn;
+  }
+  // if the value is not function then return as is
+  return value;
+}
+// Responsible for rendering any chart//
+window.FusionCharts.renderChart = (chartConfiguration) => {
+  const configAsJSObject = JSON.parse(chartConfiguration, this.parseFunction);
+  const chart = new FusionCharts(configAsJSObject);
+  chart.render();
+};
+
+// Method written for charts like time-series chart
+window.FusionCharts.setDataStore = (id, args) =>  {
+
+    var currentChart = FusionCharts(id);
+    var fusionDataStore = new FusionCharts.DataStore();
+    
+    var data = JSON.parse(args[0]);
+    var schema = JSON.parse(args[1]);
+
+    var fusionTable = fusionDataStore.createDataTable(data, schema);
+
+    currentChart.setChartData({data: fusionTable});
+};
+
+// resizeTo will not return any data due to circular json object
```

Comment:
Please add comment as to why we need a resizeTo function while we have a generic function?

Reply:
I think this function also will not be needed with the resizeTo method is called on the 'loaded' event of the chart

Reply:
So, you're saying we can call resizeTo with generic method only?